cnd/src/http_api/routes/index.rs

```diff
@@ -96,24 +96,45 @@ pub async fn post_herc20_halight_bitcoin(
         .map_err(warp::reject::custom)
 }
 
-// `warp::reply::Json` is used as a return type to please the compiler
-// until proper logic is implemented
 #[allow(clippy::needless_pass_by_value)]
 pub async fn post_halight_bitcoin_herc20(
     body: serde_json::Value,
-    _facade: Facade,
-) -> Result<warp::reply::Json, Rejection> {
-    let _body = Body::<HalightLightningBitcoin, Herc20EthereumErc20>::deserialize(&body)
+    facade: Facade,
+) -> Result<impl Reply, Rejection> {
+    let body = Body::<HalightLightningBitcoin, Herc20EthereumErc20>::deserialize(&body)
         .map_err(anyhow::Error::new)
         .map_err(problem::from_anyhow)
         .map_err(warp::reject::custom)?;
 
-    tracing::error!("Lightning routes are not yet supported");
-    Err(warp::reject::custom(
-        HttpApiProblem::new("Route not yet supported.")
-            .set_status(StatusCode::BAD_REQUEST)
-            .set_detail("This route is not yet supported."),
-    ))
+    let swap_id = LocalSwapId::default();
+    let reply = warp::reply::reply();
+
+    let swap = body.to_created_swap(swap_id);
+    facade
+        .save(swap)
+        .await
+        .map_err(problem::from_anyhow)
+        .map_err(warp::reject::custom)?;
+
+    let identities = Identities {
+        ethereum_identity: Some(body.beta.identity),
+        lightning_identity: Some(body.alpha.identity),
+    };
+    let digest = swap_digest::Herc20Halight::from(body.clone()).digest();
```

Comment:
This is buggy (but doesn't need changing IMO). This digest can conflict with a digest for a swap with the same details but alpha/beta inverted i.e., herc20<->halight. @bonomat has stared on a PR to fix this.

Reply:
Oh, true. Given the docstring I actually thought `swap_digest::Herc20Halight` was supposed to work for both directions (I still interpret the `<->` as "in both directions"). But from your explanation I also view it as a bug.